submissions/SofiiaTrokhymchuk/frogger-game/app.js

```diff
@@ -0,0 +1,152 @@
+const TILE_WIDTH = 101;
+const TILE_HEIGHT = 83;
+
+const ROWS = 5;
+const COLUMNS = 4;
+
+const BOARD_WIDTH = TILE_WIDTH * COLUMNS;
+const BOARD_HEIGHT = TILE_HEIGHT * ROWS;
+
+const CHARACTERS_WIDTH = 101;
+const CHARACHTERS_HEIGHT = 171;
+
+const CHARACTERS_SIZE_DIFFERENCE = 20;
+
+const INITIAL_ENEMY_X = -CHARACTERS_WIDTH;
+const ENEMY_MIN_SPEED = 100;
+const ENEMY_MAX_SPEED = 500;
+
+const INITAL_PLAYER_X = BOARD_WIDTH / Math.ceil(COLUMNS / 2);
+const INITAL_PLAYER_Y = BOARD_HEIGHT - CHARACTERS_SIZE_DIFFERENCE;
+
+const Enemy = function(x, y, player) {
+    this.sprite = 'images/enemy-bug.png';
+    this.x = x;
+    this.y = y;
+    this.speed = this.generateSpeed();
+    this.player = player;
+};
+
+Enemy.prototype.update = function(dt) {
+    this.checkCollisions();
+    this.x += this.speed * dt;
+    if(this.x > BOARD_WIDTH + CHARACTERS_WIDTH){
+        this.x = INITIAL_ENEMY_X;
+    };
+};
+
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Enemy.prototype.generateSpeed = function() {
+    return (Math.floor(Math.random() * (ENEMY_MAX_SPEED / 100 - ENEMY_MIN_SPEED / 100 + 1)) + ENEMY_MIN_SPEED / 100) * 100;
+};
+
+Enemy.prototype.checkCollisions = function() {
+    if(Math.abs(this.x - this.player.x) <= CHARACTERS_WIDTH - CHARACTERS_SIZE_DIFFERENCE * 1.5 
+    && Math.abs(this.y - this.player.y) <= CHARACTERS_SIZE_DIFFERENCE){
+        this.player.loses += 1;
+        this.player.addWinLossPointsToBlock(this.player.pointsBlock, this.player);
+        this.player.resetInitialPosition();
+    }
+};
+
+function generateEnemies(player){
+    return [2, 3, 4].map(rowNumber => new Enemy(INITIAL_ENEMY_X, 
+        TILE_HEIGHT * rowNumber - (CHARACHTERS_HEIGHT / 2 + CHARACTERS_SIZE_DIFFERENCE), player));
+};
```

Comment:
Why is it defined here? It works. Although as a peer developer I'd expect to find this function somewhere closer to the place it is used and after `player` is defined.